x-pack/plugins/enterprise_search/public/applications/shared/query_params/query_params.test.ts

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import queryString from 'query-string';
+
+import { parseQueryParams } from './';
+
+describe('parseQueryParams', () => {
+  it('should call queryString parse method', () => {
+    const parseMock = jest.fn();
+    jest.spyOn(queryString, 'parse').mockImplementation(parseMock);
+    parseQueryParams('?foo=bar');
+
+    expect(queryString.parse).toHaveBeenCalledWith('?foo=bar', { arrayFormat: 'bracket' });
+  });
```

Comment:
Sorry, I'm probably being too vague in my above comments, here's specifically the kind of test I'm looking for over using jest spies/mocks:

```suggestion
  it('parse query strings', () => {
    expect(parseQueryParams('?foo=bar')).toEqual({ foo: 'bar' });
    expect(parseQueryParams('?foo[]=bar&foo[]=baz')).toEqual({ foo: ['bar', 'baz'] });
  });
```

I strongly feel it's more useful from a dev understanding POV than `expected [...] toHaveBeenCalled` because it gives you actual input/output understanding of what the function does.

Reply:
Scotty, you're too fast!! Marking this resolved